### ISSUE TITLE
Shouldn't look at the members of a river relation if the relation already has the wikidata

### ISSUE THREAD

**Issue by rory** [EdwardBetts/osm-wikidata#90]
Consider the River Dodder: https://osm.wikidata.link/Q2055328 The [waterway relation in OSM](https://www.openstreetmap.org/relation/3609774) has wikidata tags, but each of the individual ways that make it up also have the `name=Dodder`. This tool is suggesting the ways as possible match ups.

I don't think it should do this. If there is a river with the wikidata tag, then it shouldn't look at/suggest the compenent ways.

*(I'm not sure those OSM ways should have the name tag if they are part of the relation...)*

**Comment by EdwardBetts** [comment]
Good catch. The matcher is not good with rivers because osm2pgsql doesn't support unclosed relations. I'll remove rivers from the match results.

**Comment by rory** [comment]
If you import with `--slim` you'll get `planet_osm_rels` which will have all the relations. it's a bit messy to query, but you can do it.

**Comment by ivanbranco** [comment]
Hi, the issue is still valid. Maybe since 2017 something changed and now can be fixed?

**Comment by tuukka** [comment]
At least both the new `flex` output mode as well as the old `pgsql` output mode of osm2pgsql now support Lua callbacks to customise how relations are handled:
* `flex` https://osm2pgsql.org/doc/manual.html#the-flex-output
  * How to copy the tags from any relation to its member ways: https://osm2pgsql.org/doc/manual.html#stages
  * Example regarding route relations: https://github.com/openstreetmap/osm2pgsql/blob/master/flex-config/route-relations.lua
* `pgsql` https://osm2pgsql.org/doc/manual.html#the-pgsql-output
  * Example to tweak how relations are handled, also mentions route relations: https://github.com/openstreetmap/osm2pgsql/blob/master/style.lua

**Comment by kjonosm** [comment]
Hi, the issue is still valid. Maybe with all the updates in osm2pgsql lately something changed and now can be fixed?
